cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -662,14 +698,18 @@ private List<ApplicationId> decodeAndValidateBatchApplicationRecord(NamespaceId
           throw new BadRequestException("Missing 'name' in the request element for app-id.");
         }
         if (element.getAppVersion() == null) {
-          appIds.add(validateApplicationId(namespaceId, element.getName()));
+          // fetch the latest version
+          String latestAppVersion = getLatestAppVersion(validateApplicationId(namespaceId, element.getName()));
+          appIds.add(validateApplicationVersionId(namespaceId, element.getName(), latestAppVersion));
         } else {
           appIds.add(validateApplicationVersionId(namespaceId, element.getName(), element.getAppVersion()));
         }
       }
       return appIds;
     } catch (JsonSyntaxException e) {
       throw new BadRequestException("Request body is invalid json: " + e.getMessage());
+    } catch (Exception e) {
+      throw new BadRequestException("Could not find app " + e.getMessage());
```

Comment:
Why "could not find"? This is a very bold statement that any exception means "could not find". Also for "could not find" we should produce a proper exception to generate 404

Reply:
So I think we can change the response to be - Invalid app-id provided or something like that - this is thrown when the user either enters an invalid app-name or app-version in the request

Reply:
Overall you should not try hard to transform exceptions in the handler. Just throw the message out and exception handler will handle it. Only if you need some specific different Http code, you need to transform exceptions